pkg/admission/permissionclaims/mutating_permission_claims.go

```diff
@@ -93,29 +96,31 @@ func (m *mutatingPermissionClaims) Admit(ctx context.Context, a admission.Attrib
 	}
 
 	labels := u.GetLabels()
-	if labels == nil {
-		labels = make(map[string]string)
-	}
-
-	// Start by removing all claim labels
 	for k := range labels {
-		if strings.HasPrefix(k, apisv1alpha1.APIExportPermissionClaimLabelPrefix) {
+		if !strings.HasPrefix(k, apisv1alpha1.APIExportPermissionClaimLabelPrefix) {
+			continue
+		}
+		if _, expected := expectedLabels[k]; !expected {
 			delete(labels, k)
 		}
 	}
-
-	// Add back what should be there
 	for k, v := range expectedLabels {
+		if labels == nil {
```

Comment:
Why do this inside the loop instead of at line 96?

Reply:
because only here it is needed